### PR TITLE
feat(core): new enabledO365MailForward virtual attribute

### PR DESCRIPTION
**Pull request by kofzera**
Added a new virt attribute. The value is the same as o365MailForward attribute if disableo365MailForward is not true, empty string otherwise.

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForward.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+public class urn_perun_user_facility_attribute_def_virt_enabledO365MailForward extends UserFacilityVirtualAttributesModuleAbstract implements UserFacilityVirtualAttributesModuleImplApi {
+
+	@Override
+	public Attribute getAttributeValue(PerunSessionImpl sess, User user, Facility facility, AttributeDefinition attributeDefinition) {
+		Attribute attr = new Attribute(attributeDefinition);
+		try {
+			Attribute disableForwardAtrr = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, facility, user,
+				AttributesManager.NS_USER_FACILITY_ATTR_VIRT + ":disableO365MailForward");
+			if (disableForwardAtrr.getValue() != null && !(Boolean) disableForwardAtrr.getValue()) {
```
Comment (xflord) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForward.java:24]:
```suggestion
			if (disableForwardAtrr.getValue() != null && !disableForwardAtrr.valueAsBoolean()) {
```

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForward.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+				Attribute o365MailForwardAttr = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, facility, user,
+					AttributesManager.NS_USER_FACILITY_ATTR_VIRT + ":o365MailForward");
+				String o365MailForward = (String) o365MailForwardAttr.getValue();
```
Comment (xflord) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForward.java:27]:
```suggestion
				String o365MailForward = o365MailForwardAttr.valueAsString();
```

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForward.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ * Otherwise, the value of o365MailForward Attribute is returned.
+ */
+public class urn_perun_user_facility_attribute_def_virt_enabledO365MailForward extends UserFacilityVirtualAttributesModuleAbstract implements UserFacilityVirtualAttributesModuleImplApi {
+
+	@Override
+	public Attribute getAttributeValue(PerunSessionImpl sess, User user, Facility facility, AttributeDefinition attributeDefinition) {
+		Attribute attr = new Attribute(attributeDefinition);
+		try {
+			Attribute disableForwardAtrr = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, facility, user,
+				AttributesManager.NS_USER_FACILITY_ATTR_VIRT + ":disableO365MailForward");
```
Comment (xflord) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForward.java:23]: Please extract the related attribute names to constants for readability

File: perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForwardTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+		disabledMailFwdAttr.setValue(true);
+		o365MailAttr.setValue(exampleMail);
+	}
+
+	@Test
+	public void getAttributeValueWithEnabledFwdTest() throws Exception {
+		System.out.println("urn_perun_user_facility_attribute_def_virt_enabledO365MailForward.GetAttributeValue()");
+
+		when(session.getPerunBl().getAttributesManagerBl().getAttribute(any(PerunSessionImpl.class), any(Facility.class), any(User.class), eq(AttributesManager.NS_USER_FACILITY_ATTR_VIRT + ":disableO365MailForward"))).thenReturn(enabledMailFwdAttr);
+		when(session.getPerunBl().getAttributesManagerBl().getAttribute(any(PerunSessionImpl.class), any(Facility.class), any(User.class), eq(AttributesManager.NS_USER_FACILITY_ATTR_VIRT + ":o365MailForward"))).thenReturn(o365MailAttr);
```
Comment (xflord) [inline perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForwardTest.java:49]: move this to the setup method

File: perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForwardTest.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.*;
```
Comment (xflord) [inline perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForwardTest.java:3]: Make sure you disable wildcard imports in your IDE

**Review by xflord — COMMENTED** [review]
Also implement the `getStrongDependencies()` method (returning the two related attributes)

File: perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForwardTest.java (hunk 4 above)
Comment (kofzera) [reply perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForwardTest.java]: Moved only the second line as the first one differs between the methods.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForward.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.*;
```
Comment (bodnara) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_facility_attribute_def_virt_enabledO365MailForward.java:3]: Remove wildcard import.

**Review by bodnara — CHANGES_REQUESTED** [review]

**Review by bodnara — APPROVED** [review]

**Pull request merged**